### PR TITLE
Remove SSAA checkbox and add SSAA to AA dropdown

### PR DESCRIPTION
**Pull request by JosJuice**
The checkbox is removed, and the dropdown now looks like this:

~~None
2x MSAA
2x SSAA
4x MSAA
4x SSAA
6x MSAA
6x SSAA
8x MSAA
8x SSAA~~

EDIT:
None
2x MSAA
4x MSAA
8x MSAA
2x SSAA
4x SSAA
8x SSAA

### CODE REVIEW

File: Source/Core/DolphinWX/VideoConfigDiag.cpp [hunk 1]
```diff
@@ -751,24 +742,38 @@ void VideoConfigDiag::PopulatePostProcessingShaders()
...
 		else
-			choice_aamode->AppendString(std::to_string(mode) + suffix);
+		{
+			const std::string mode_string = std::to_string(mode);
+			choice_aamode->AppendString(mode_string + "x MSAA");
+			choice_aamode->AppendString(mode_string + "x SSAA");
+		}
+	}
+
+	int selected_mode = vconfig.iMultisampleMode;
```
Comment (degasus) [inline Source/Core/DolphinWX/VideoConfigDiag.cpp]: Do you want to replace the vconfig.iMultisampleMode global variable with the amount of samples itself?
Always using aamodes[iMultisampleMode] is scary.

Comment (JosJuice) [inline Source/Core/DolphinWX/VideoConfigDiag.cpp]: I don't know how it should be handled in the backend and config. Are you asking about that, or are you asking about problems that might occur with `choice_aamode->SetSelection(selected_mode)`?

File: Source/Core/DolphinWX/VideoConfigDiag.cpp [hunk 2]
```diff
@@ -751,24 +742,38 @@ void VideoConfigDiag::PopulatePostProcessingShaders()
...
 		if (mode == 1)
+		{
 			choice_aamode->AppendString(_("None"));
+		}
 		else
-			choice_aamode->AppendString(std::to_string(mode) + suffix);
+		{
+			const std::string mode_string = std::to_string(mode);
+			choice_aamode->AppendString(mode_string + "x MSAA");
+			choice_aamode->AppendString(mode_string + "x SSAA");
```
Comment (degasus) [inline Source/Core/DolphinWX/VideoConfigDiag.cpp]: if(vconfig.backend_info.bSupportsSSAA)

**Comment by DonelBueno** [comment]
None
2x MSAA
4x MSAA
8x MSAA
2x SSAA
4x SSAA
8x SSAA

Don't you think this looks better and is more logical? And 6x MSAA and 6x SSAA? Are they really necessary?

**Comment by JosJuice** [comment]
I think sorting them by number is neater, but I only have a weak opinion on this. I'd appreciate if someone else can comment on it.

The OpenGL backend providing a 6x option is out of scope for this PR (and I have no idea if that option is appropriate or not because I don't have any experience in this area). Do you want me to make a separate PR for it?

**Comment by Tilka** [comment]
You should sort them by "more expensive and better looking" because that's what people expect.

**Comment by JosJuice** [comment]
Is that the sorting with all SSAA options at the end? (Sorry, I really don't know much about anti-aliasing)

**Comment by phire** [comment]
Basically, MSAA keeps the load on the pixel shaders the same while increasing load on the ROPs, Memory Bandwidth and Framebuffer size and only anti-aliases around the edges of polygons, while SSAA does the same but also increases load on pixel shaders and texture fetches (which increases memory bandwidth again) to give better anti-aliasing for partially transparent polygons.

**Comment by aserna3** [comment]
I'd like to bikeshed the SSAA options grouped together at the bottom of the list. Generally SSAA is significantly more taxing than MSAA, so it being further down shows this, as well as keeps it looking visually simple. I had a hard time visually sizing up that list.

Also, will this load/save into config properly? At first glance it looks like it should. If it does then LGTM

**Comment by JosJuice** [comment]
It turns out that 6x was just a typing mistake from my side. It doesn't exist in Dolphin.

> I'd like to bikeshed the SSAA options grouped together at the bottom of the list.

I'm working on it already. It'll be pushed soon.

> Also, will this load/save into config properly?

Of course. What's the point of the config window otherwise? :P

**Comment by aserna3** [comment]
I guess I'm just curious because that was the obstacle I was trying to work around when I attempted tackling this a month ago. I couldn't think of how to not break config while at the same time loading the correct options, and you did it in 40~ lines of code. Nice work. :3

**Comment by JosJuice** [comment]
Updated. All SSAA modes are now at the bottom.

**Comment by MaJoRoesch** [comment]
Screen shots please!

**Comment by MaJoRoesch** [comment]
Screen shots please!

(It's an artist thing, I like to see...)

**Comment by JosJuice** [comment]
(The image is edited so that the dropdown and description are visible at the same time.)

**Comment by phire** [comment]
I think the extra performance requirements of MSAA are much smaller than your description is implying.

MSAA for something like dolphin is more or less free on dedicated GPUs.

**Comment by MaJoRoesch** [comment]
@phire that depends on the game. I've had times where 8xMSAA *is* too demanding! Don't underestimate it. The only free antialiasing is FXAA, and it's not so great.

**Comment by degasus** [comment]
@MaJoRoesch Lots of GPUs don't support 8x MSAA, so internally, it will be 4x MSAA + 2x SSAA. Just stay at 4x MSAA to be sure to not get the higher pixel shader effort.

**Comment by phire** [comment]
@degasus which brings up another question (outside the scope of this PR)

Do we want the options to say exactly what form of AA you are getting from your graphics card, like if the gpu is actually going to do 4x MSAA + 2x SSAA (potentially actually combined with temporal dithering), we should actually say that.

**Comment by JosJuice** [comment]
The description has been changed to MaJoR's suggestion, except I replaced "also applies AA" with "also applies anti-aliasing".

**Comment by aserna3** [comment]
LGTM. Merge it! :3

**Comment by fallaha56** [comment]
guys hope this isn't a bad place for this (figure it's folk who are interested in AA here) but there's a major unresolved bug with D3D antialiasing -there's some kind of scaling going on in ?backbuffers, leads to a hoorible staircasing effect

very easy to see in F-Zero on the tracks (and not in OpenGL)

**Comment by JosJuice** [comment]
This is the wrong place. You can submit a bug report here if you want to: https://bugs.dolphin-emu.org/projects/emulator

**Comment by fallaha56** [comment]
thanks! will do. any help with this much appreciated, OpenGL does not run well for me at all

**Pull request merged**